packages/app-form-builder/src/components/Form/FormRender.tsx

```diff
@@ -73,8 +89,12 @@ const FormRender: React.FC<FbFormRenderComponentProps> = props => {
         return fields.find(field => field.fieldId === id) || null;
     };
 
-    const getFields = (): FormRenderFbFormModelField[][] => {
-        const fieldLayout = cloneDeep(layout);
+    // We need to have "stepIndex" prop in order to get corresponding fields for the current step.
+    const getFields = (stepIndex: number): FormRenderFbFormModelField[][] => {
+        // We need this check in case we deleted last step and at the same time we were previewing it.
+        const stepFields =
+            steps[stepIndex] === undefined ? steps[steps.length - 1] : steps[stepIndex];
```

Comment:
Wouldn't the current step just be reset when the user deletes the step? So, in the preview tab, we'd just start from the beginning.

If the above assumption is correct, do we need this code here?

Reply:
We need this block of code because of the way useState works. It changes `currentStep` value with a delay and because of that we get undefined.